cmd/kube-scheduler/app/server.go

```diff
@@ -212,7 +212,11 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 	if !cc.ComponentConfig.DelayCacheUntilActive || cc.LeaderElection == nil {
 		startInformersAndWaitForSync(ctx)
 	}
-
+	// Wait for all handlers to sync (all items in the initial list delivered) before scheduling.
+	if err := sched.WaitForHandlersSync(ctx); err != nil {
```

Comment:
Shouldn't this be part of the `startInformersAndWaitForSync`? We'll also sync the cache in the leader elector mode.